core/utils_test.py

```diff
@@ -794,3 +794,85 @@ def test_url_open(self) -> None:
+        with self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            AssertionError, 'Only positive integers allowed'):
+            utils.get_random_int(-1)
+
+    def test_get_random_choice(self) -> None:
+        list_instance = [1, 5, 9, 11, 15]
+        list_instance2: List[str] = []
+        self.assertIn(utils.get_random_choice(list_instance), list_instance)
+        with self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            AssertionError, 'Only non-empty lists allowed'):
+            utils.get_random_choice(list_instance2)
+
+    def test_get_human_readable_time_string(self) -> None:
+        """Should we test for negative time."""
+        self.assertEqual('December 12 06:42:12', utils.get_human_readable_time_string(944980932342.38))  # pylint: disable=line-too-long
+
+    def test_generate_new_session_id(self) -> None:
+        self.assertEqual(24, len(utils.generate_new_session_id()))
+        self.assertEqual(type(utils.generate_new_session_id()).__name__, 'str')
+
+    def test_require_valid_name_with_incorrect_input(self) -> None:
+        """ADD TEST ESCAPE SEQUENCE CHARACTERS CHECK."""
+        with self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            utils.ValidationError, 'The length of the exploration title should be between 1 and 50 ' 'characters; received '''):   # pylint: disable=line-too-long
+            utils.require_valid_name('', 'the exploration title')
+        with self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            utils.ValidationError, 'Names should not start or end with whitespace.'):  # pylint: disable=line-too-long
+            utils.require_valid_name(' 123\n', 'the exploration title')
+        with self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            utils.ValidationError, 'Adjacent whitespace in the exploration title should be collapsed.'):  # pylint: disable=line-too-long
+            utils.require_valid_name('1  23', 'the exploration title')
+        with self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            utils.ValidationError, 'Invalid character : in the exploration title: 1\n:23'):  # pylint: disable=line-too-long
+            utils.require_valid_name('1\n:23', 'the exploration title')
+
+    def test_get_hex_color_for_category(self) -> None:
+        self.assertEqual(
+            utils.get_hex_color_for_category('Law'), '#538270')
+        self.assertEqual(
+            utils.get_hex_color_for_category('Quantum Physics'), '#a33f40')
+
+    def test_unescape_encoded_uri_component(self) -> None:
+        self.assertEqual(utils.unescape_encoded_uri_component('/El%20Ni%C3%B1o/'), '/El Niño/')  # pylint: disable=line-too-long
+
+    def test_compress_and_decompress_zlib(self) -> None:
+        string_instance = b'This is for testing'
+        string_compressed = utils.compress_to_zlib(string_instance)
+        self.assertNotEqual(len(string_compressed), len(string_instance))
```

Comment:
Shouldn't the compressed string be shorter than the original and not just of a different length?

Reply:
Ahh there is a mistake here, compressed string will not reduce the length of the string but rather compresses the **data** required by the string. So we should be comparing the data stored by the strings and not their lengths.

Reply:
I will make the necessary changes

Reply:
https://stackoverflow.com/questions/49432661/zlib-compress-in-python
This is the article I referred to